pkg/controller/elasticsearch/remotecluster/elasticsearch.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package remotecluster
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/license"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/tracing"
+	esclient "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/services"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"go.elastic.co/apm"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("remotecluster")
+
+// UpdateRemoteCluster updates the remote clusters in the persistent settings by calling the Elasticsearch API.
+func UpdateRemoteCluster(
+	ctx context.Context,
+	c k8s.Client,
+	esClient esclient.Client,
+	licenseChecker license.Checker,
+	es esv1.Elasticsearch,
+) error {
+	span, _ := apm.StartSpan(ctx, "update_remote_clusters", tracing.SpanTypeApp)
+	defer span.End()
+
+	enabled, err := licenseChecker.EnterpriseFeaturesEnabled()
+	if err != nil {
+		return err
+	}
+	if !enabled {
+		log.Info(
+			"Remote cluster is an enterprise feature. Enterprise features are disabled",
+			"namespace", es.Namespace, "es_name", es.Name,
+		)
+		return nil
+	}
+
+	currentRemoteClusters, err := getCurrentRemoteClusters(es)
+	if err != nil {
+		return err
+	}
+	if currentRemoteClusters == nil {
+		currentRemoteClusters = make(map[string]string)
+	}
+	expectedRemoteClusters := getExpectedRemoteClusters(es)
+
+	updated := false
+	// RemoteClusters to add or update
+	for name, remoteCluster := range expectedRemoteClusters {
+		if currentConfigHash, ok := currentRemoteClusters[name]; !ok || currentConfigHash != remoteCluster.ConfigHash {
+			// Declare remote cluster in ES
+			seedHosts := []string{services.ExternalTransportServiceHostname(remoteCluster.ElasticsearchRef.NamespacedName())}
+			persistentSettings := newRemoteClusterSetting(name, seedHosts)
+			log.Info("Add or update remote cluster",
+				"namespace", es.Namespace,
+				"es_name", es.Name,
+				"remote_cluster", remoteCluster.Name,
+				"seeds", seedHosts,
+			)
+			if err := updateRemoteCluster(esClient, persistentSettings); err != nil {
+				return err
+			}
+			updated = true
+		}
+	}
+
+	// RemoteClusters to remove
+	for name := range currentRemoteClusters {
+		if _, ok := expectedRemoteClusters[name]; !ok {
+			persistentSettings := newRemoteClusterSetting(name, nil)
+			log.Info("Remove remote cluster",
+				"namespace", es.Namespace,
+				"es_name", es.Name,
+				"remote_cluster", name,
+			)
+			err := updateRemoteCluster(esClient, persistentSettings)
```

Comment:
Should we not try to make this just one HTTP call for all added/removed remote clusters?